selfdrive/car/mazda/carcontroller.py

```diff
@@ -0,0 +1,44 @@
+from selfdrive.car.mazda import mazdacan
+from selfdrive.car.mazda.values import SteerLimitParams, Buttons
+from opendbc.can.packer import CANPacker
+from selfdrive.car import apply_std_steer_torque_limits
+
+class CarController():
+  def __init__(self, dbc_name, CP, VM):
+    self.apply_steer_last = 0
+    self.packer = CANPacker(dbc_name)
+    self.steer_rate_limited = False
+
+  def update(self, enabled, CS, frame, actuators, pcm_cancel_cmd, visual_alert,
+             left_lane, right_lane, left_lane_depart, right_lane_depart):
+    """ Controls thread """
+
+    can_sends = []
+
+    ### STEER ###
+
+    if enabled:
+      # calculate steer and also set limits due to driver torque
+      new_steer = int(round(actuators.steer * SteerLimitParams.STEER_MAX))
+      apply_steer = apply_std_steer_torque_limits(new_steer, self.apply_steer_last,
+                                                  CS.out.steeringTorque, SteerLimitParams)
+      self.steer_rate_limited = new_steer != apply_steer
+
+      if CS.out.standstill and frame % 50 == 0:
+        # Mazda Stop and Go requires a RES button (or gas) press if the car stops more than 3 seconds
+        # Send Resume button at 2hz if we're engaged at standstill to support full stop and go!
+        # TODO: improve the resume trigger logic by looking at actual radar data
+        can_sends.append(mazdacan.create_cancel_acc(self.packer, CS.CP.carFingerprint, Buttons.RESUME))
+    else:
```

Comment:
This should be like the other cars where we cancel on `pcm_cancel_cmd`

Reply:
Sometimes we may need to send a couple of msgs to get acc cancel, would switching to pcm_cancel_cmd cause an issue there?

Reply:
No, it should behave the same in that regard